File: 2001-2500/2062.Count Vowel Substrings of a String.ts
function countVowelSubstrings(word: string): number {
    const vowels = 'aeiou'.split('');
    const cnt = {};
    vowels.map(c => cnt[c] = 0);
    let res = 0, preStart = 0, l = 0;
    for (let r = 0; r <= word.length; r++) {
        if (r < word.length && 'aeiou'.includes(word[r])) {
            cnt[word[r]]++;
            while (cnt[word[l]] > 1) {
                cnt[word[l++]]--;
            }
            if (Object.values(cnt).filter(v => v).length == vowels.length) {
                res += l - preStart + 1;
            }
        } else {
            vowels.map(c => cnt[c] = 0);
            preStart = l = r + 1;
        }
    }
    return res;
};

File: 2001-2500/2062.Count Vowel Substrings of a String.py
class Solution(object):
    def countVowelSubstrings(self, word):
        """
        :type word: str
        :rtype: int
        """
        res = 0
        for start in range(len(word)):
            for end in range(start + 1, len(word) + 1):
                if set(word[start:end]) == set("aeiou"):
                    res += 1
        return res

